react/features/overlay/reducer.js

```diff
@@ -0,0 +1,140 @@
+
+    case SUSPEND_DETECTED:
+        return _suspendDetected(state, action);
+    }
+
+    return state;
+});
+
+/**
+ * Reduces a specific Redux action CONFERENCE_FAILED of the feature overlay.
+ *
+ * @param {Object} state - The Redux state of the feature overlay.
+ * @param {Action} action - The Redux action CONFERENCE_FAILED to reduce.
+ * @returns {Object} The new state of the feature overlay after the reduction of
+ * the specified action.
+ * @private
+ */
+function _conferenceFailed(state, action) {
+    const ConferenceErrors = JitsiMeetJS.errors.conference;
+
+    if (action.error === ConferenceErrors.FOCUS_LEFT
+            || action.error === ConferenceErrors.VIDEOBRIDGE_NOT_AVAILABLE) {
+        return setStateProperties(state, {
+            haveToReload: true,
+            isNetworkFailure: false,
+            reason: action.errorMessage
+        });
+    }
+
+    return state;
+}
+
+/**
+ * Reduces a specific Redux action CONNECTION_ESTABLISHED of the feature
+ * overlay.
+ *
+ * @param {Object} state - The Redux state of the feature overlay.
+ * @returns {Object} The new state of the feature overlay after the reduction of
+ * the specified action.
+ * @private
+ */
+function _connectionEstablished(state) {
+    return setStateProperty(state, 'connectionEstablished', true);
+}
+
+/**
+ * Reduces a specific Redux action CONNECTION_FAILED of the feature overlay.
+ *
+ * @param {Object} state - The Redux state of the feature overlay.
+ * @param {Action} action - The Redux action CONNECTION_FAILED to reduce.
+ * @returns {Object} The new state of the feature overlay after the reduction of
+ * the specified action.
+ * @private
+ */
+function _connectionFailed(state, action) {
+    const ConnectionErrors = JitsiMeetJS.errors.connection;
+
+    switch (action.error) {
+    case ConnectionErrors.CONNECTION_DROPPED_ERROR:
+    case ConnectionErrors.OTHER_ERROR:
+    case ConnectionErrors.SERVER_ERROR:
+        logger.error(`XMPP connection error: ${action.errorMessage}`);
+
+        // From all of the cases above only CONNECTION_DROPPED_ERROR
+        // is considered a network type of failure
+        return setStateProperties(state, {
+            haveToReload: true,
+            isNetworkFailure:
+                action.error === ConnectionErrors.CONNECTION_DROPPED_ERROR,
+            reason: `xmpp-conn-dropped: ${action.errorMessage}`
+        });
+    }
+
+    return state;
+}
+
+/**
+ * Reduces a specific Redux action MEDIA_PERMISSION_PROMPT_VISIBILITY_CHANGED of
+ * the feature overlay.
+ *
+ * @param {Object} state - The Redux state of the feature overlay.
+ * @param {Action} action - The Redux action to reduce.
+ * @returns {Object} The new state of the feature overlay after the reduction of
+ * the specified action.
+ * @private
+ */
+function _mediaPermissionPromptVisibilityChanged(state, action) {
+    return setStateProperties(state, {
+        mediaPermissionPromptVisible: action.isVisible,
```

Comment:
I don't have any preferences about a rule for naming boolean variables and in the PR simply there is no rule about it. 

Do we have a rule for this? Sorry if there was a rule that I broke.